x/stake/handler.go

```diff
@@ -31,7 +31,10 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
 }
 
 // Called every block, update validator set
-func EndBlocker(ctx sdk.Context, k keeper.Keeper) (ValidatorUpdates []abci.ValidatorUpdate) {
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) (
+	TendermintUpdates []abci.ValidatorUpdate,
+	ValidatorUpdates []sdk.ValidatorUpdate, totalPower int64) {
```

Comment:
This is weird, can't we derive TendermintUpdates from ValidatorUpdates?  If so then we should do that, this introduces more surface area for bugs where they aren't updated in synchrony.

Reply:
Yeah well we can - it's just annoying to have to convert validator.Pubkey -> Tendermint-Compatible-Pubkey then back to sdk.ConsAddress for the distr.store - realized that it may actually be better to be using an sdk.ValAddress for the distr.store too - I _would_ like to combine the output `TendermintUpdates` and `ValidatorUpdates` into one return object, I just don't want to have to loop through that object again once it's received within distribution... does that make sense? Have any ideas?

Reply:
mmm, I don't think we even need this!  as per my other comment https://github.com/cosmos/cosmos-sdk/pull/2541/files#r226873997